codice-itest-impl/src/main/java/org/codice/itest/SkipAwareExecutorService.java

```diff
@@ -0,0 +1,48 @@
+package org.codice.itest;
+
+import org.codice.itest.api.TestResult;
+import org.codice.itest.api.TestResultFactory;
+import org.codice.itest.reporter.RemainingTestsReporter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Arrays;
+import java.util.concurrent.*;
+import java.util.function.Consumer;
+
+import static org.codice.itest.api.TestStatus.NOT_EXECUTED;
+
+@Component("SkipAwareExecutorService")
+public class SkipAwareExecutorService extends ThreadPoolExecutor {
```

Comment:
:question: Can this logic be woven into the existing executor?  This seems like a lot of duplication for not a lot of additional logic.

Reply:
As far as we can tell we need to extend it in order to access that part of the execution, but we are open to any ways you guys can think of.

Reply:
I think there may be a way to do this with the existing executor. I'm thinking the following snippet would go here: https://github.com/codice/codice-itest/blob/master/codice-itest-impl/src/main/java/org/codice/itest/IntegrationTestService.java#L77

The `awaitTermination` method blocks until all threads finish or time out. It returns a boolean if the timeout was exceeded, so could call the `RemainingTestsReporter` manually  in that scenario.
```
        // The timeout was exceeded
        if (!hasTerminated) {
            remainingTestReporter.getRemainingList().forEach(testName -> {
                TestResult testResult = testResultFactory.notExecuted(testName);
                testResultListenerList.forEach(listener -> listener.accept(testResult));
            });
            executorService.shutdownNow();
        }
```
Disclaimer, I haven't tried this, but it seems like it works based on the ExecutorService javadoc.

Reply:
Ok nice, I will make the adjustments.